Sources/SwiftGenKit/Parsers/ParserOptionList.swift

```diff
@@ -0,0 +1,44 @@
+//
+// SwiftGenKit
+// Copyright © 2019 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+
+public struct ParserOptionList {
+  enum Error: Swift.Error {
+    case unknownOption(key: String, value: Any)
+  }
+
+  private let options: [AnyParserOption]
+
+  init(options: [AnyParserOption] = []) {
+    self.options = options
+  }
+
+  init(lists: [ParserOptionList]) {
+    self.options = lists.flatMap { $0.options }
+  }
+
+  /// Check if the given dictionary of key values are all supported options.
+  ///
+  /// - Parameter options: a dictionary of options
+  func check(options: [String: Any]) throws {
+    let known = Set(self.options.map { $0.key })
+    for option in options where !known.contains(option.key) {
+      throw Error.unknownOption(key: option.key, value: option.value)
+    }
```

Comment:
The method will exit at the first `throw`, so looping with that `for` is useless here imho.

You might replace it with:
```swift
    let knownKeys = Set(self.options.map { $0.key })
    if let badOption = options.first(where: { !knownKeys.contains($0.key) }) {
      throw Error.unknownOption(key: badOption.key, value: badOption.value)
    }
```